cobalt/loader/fetcher_cache_test.cc

```diff
@@ -0,0 +1,433 @@
+    if (response_type == kLoadResponseAbort) {
+      return;
+    }
+  }
+
+  void FireReceived(const char* data, size_t size) {
+    handler()->OnReceived(this, data, size);
+  }
+
+  void FireDone() { handler()->OnDone(this); }
+
+  void FireFetcherProcess(
+      const scoped_refptr<net::HttpResponseHeaders>& headers, const char* data,
+      size_t size) {
+    LoadResponseType response = handler()->OnResponseStarted(this, headers);
+    if (response == kLoadResponseContinue) {
+      handler()->OnReceived(this, data, size);
+      handler()->OnDone(this);
+    }
+  }
+};
+
+class StubFetcherHandler : public Fetcher::Handler {
+ public:
+  // From Fetcher::Handler.
+  void OnReceived(Fetcher* fetcher, const char* data, size_t size) override {}
+  void OnDone(Fetcher* fetcher) override {}
+  void OnError(Fetcher* fetcher, const std::string& error_message) override {}
+};
+
+struct StubFetcherFactory {
+ public:
+  StubFetcherFactory() {}
+  // Way to access the last fetcher created by the fetcher factory.
+  StubFetcher* fetcher;
+
+  std::unique_ptr<Fetcher> Create(Fetcher::Handler* handler) {
+    fetcher = new StubFetcher(handler);
+    return base::WrapUnique<Fetcher>(fetcher);
+  }
+
+  Loader::FetcherCreator GetFetcherCreator() {
+    return base::Bind(&StubFetcherFactory::Create, base::Unretained(this));
+  }
+};
+
+std::unique_ptr<Fetcher> CreateFecter(Fetcher::Handler* handler) {
+  StubFetcher* fetcher = new StubFetcher(handler);
+  return base::WrapUnique<Fetcher>(fetcher);
+}
+
+}  // namespace
+
+class FetcherCacheTest : public ::testing::Test {
+ protected:
+  class OngoingFetcherFactory {
+   public:
+    OngoingFetcherFactory(const char* name, size_t capacity) {
+      fetcher_cache_ = base::MakeRefCounted<FetcherCache>(name, capacity);
+    }
+    ~OngoingFetcherFactory() {
+      if (fetcher_cache_) {
+        fetcher_cache_->DestroySoon();
+      }
+    }
+
+    size_t size() const { return fetcher_cache_->size(); }
+
+    size_t capacity() const { return fetcher_cache_->capacity(); }
+
+    Loader::FetcherCreator MakeCachedFetcherCreator(
+        const GURL& url, const Loader::FetcherCreator& real_fetcher_creator) {
+      return fetcher_cache_->GetFetcherCreator(url, real_fetcher_creator);
+    }
+
+   private:
+    scoped_refptr<FetcherCache> fetcher_cache_;
+  };
+
+  const std::string data_ = "Test string! Test string! Test!!";
+};
+
+TEST_F(FetcherCacheTest, CtorAndDtor) {
+  OngoingFetcherFactory fetcher_factory("cache_name", 1024 * 1024);
+  EXPECT_EQ(1024 * 1024, fetcher_factory.capacity());
+}
+
+TEST_F(FetcherCacheTest, MakeFetcher) {
+  OngoingFetcherFactory fetcher_factory("cache_name", 1024 * 1024);
+  auto fetcher_creator = fetcher_factory.MakeCachedFetcherCreator(
+      GURL("https://example.com"), base::Bind(&CreateFecter));
+  EXPECT_EQ(0, fetcher_factory.size());
+}
+
+TEST_F(FetcherCacheTest, SunnyDay) {
+  base::test::ScopedTaskEnvironment scoped_task_environment_;
```

Comment:
You can move the scoped_task_environment_ to be a member of the test fixture, and save adding it into each individual test.

Reply:
Done.